autoPyTorch/pipeline/components/setup/network/base_network.py

```diff
@@ -106,24 +111,35 @@ def predict(self, loader: torch.utils.data.DataLoader) -> torch.Tensor:
         """
         Performs batched prediction given a loader object
         """
-        assert self.network is not None
-        self.network.eval()
-
+        if len(self.network_snapshots) == 0:
+            assert self.network is not None
+            return self._predict(network=self.network, loader=loader).cpu().numpy()
+        else:
+            # if there are network snapshots,
+            # take average of predictions of all snapshots
+            Y_snapshot_preds: List[torch.Tensor] = list()
+
+            for network in self.network_snapshots:
+                Y_snapshot_preds.append(self._predict(network, loader))
+            Y_snapshot_preds_tensor = torch.stack(Y_snapshot_preds)
+            return Y_snapshot_preds_tensor.mean(dim=0).cpu().numpy()
```

Comment:
do we need to put this mean to cpu? As we are putting the individual predictions to cpu in line 142.

Reply:
Now that you pointed it out,  I do not think so. It should be in cpu already.